### DISCUSSION TITLE
Using Guid as Id

### DISCUSSION THREAD

**Discussion by mrDenning** [CommunityToolkit/Datasync#172] [General]
Hi - I have a code base which I'd like to use the DataSync library. The codebase is fairly mature and all the primary keys are called 'Id' and are Guids. Is there any way of overriding the Id as String requirement to fit into an existing codebase and/or a guide for changing these over to String? 
The documentation suggests this isn't possible, and [says](https://communitytoolkit.github.io/Datasync/in-depth/server/index.html#create-a-model-entity-class) "The Datasync libraries or the database maintains these properties. Do not modify these properties in your own code." -

**Comment by mrDenning** [comment]
**Update**: I found [this article](https://adrianhall.github.io/azure/2023/12/22/zumo-intid/) from @adrianhall 's blog; it describes the similar issue (int instead of Guid) using the old Azure Mobile Apps library - I'm going to give it a good read and come back here if I can't get it working or needed to do something specific to get Guid working. 👍

**Reply by adrianhall** [reply]
I happen to be on holiday right now, so not checking in anywhere as frequently as before.  However, you want to read this blog post: https://apps-on-azure.net/datasync/2024/2024-11-01-datasync-part3.html - it’s part of a series, and yes - it handles int IDs instead of GUIDs, but it’s the same concept.  It’s just written for the Community Toolkit.

**Reply by mrDenning** [reply]
Awesome - thanks! Thanks for taking the time to reply during your holiday - have a restful time.

**Comment by mrDenning** [comment]
Good article - thanks. I'm going to close this now as it provides a good guide on the matter.

**Comment by MysterDru** [comment]
Its worth noting that the https://apps-on-azure.net url doesn't work anymore. @adrianhall I don't suppose there is a back-up of these articles somewhere for reference is there?

Otherwise, here's the wayback machine url for future reference: https://web.archive.org/web/20250125000049/https://apps-on-azure.net/datasync/2024/2024-11-01-datasync-part3.html

**Reply by adrianhall** [reply]
I just wasn't getting readership, so I shut it down.

I moved the datasync articles before I shut it down.  They are available in the documentation of the toolkit.
